### ISSUE TITLE
The Ability to Use Custom Model Form Fields in Wagtail Admin Forms

### ISSUE THREAD

**Issue by MrKeals** [wagtail/guide#350]
Custom model form fields in Wagtail Admin Forms are intriguing. They offer adaptability in web development. This can be crucial for tailored solutions.

To start, understand 'why.' It's about specificity. When standard fields fall short, custom fields are invaluable. They give forms a unique touch, aligning with project needs.

Now, uncover 'how.' In Wagtail, create a custom model form, define fields, and integrate it into Wagtail Admin. Shape the form to your vision.

Using non-standard fields can be beneficial. Diverging from the norm adds uniqueness.

Ultimately, 'why' and 'how' blend in crafting a functional, custom Wagtail Admin form. Custom fields are key for distinct projects.

**Comment by MrKeals** [comment]
The Ability to Use Custom Model Form Fields in Wagtail Admin Forms
Overview:
Wagtail allows you to use custom model form fields in Wagtail admin forms. This can be useful for creating more complex or customized user interfaces for editing Wagtail models.
Benefits:
Using custom model form fields can provide a number of benefits, including:
Improved user experience: You can create more user-friendly and intuitive forms by using custom form fields.
Increased flexibility: You can use custom form fields to add new functionality to your Wagtail admin forms, such as the ability to upload files or select multiple items from a dropdown list.
Reduced development time: Using custom form fields can save you time and effort when developing new Wagtail models or extending existing models.
How to use:
To use custom model form fields in Wagtail admin forms, you need to create a subclass of the wagtail.admin.forms.WagtailAdminModelForm class. Your subclass should override the get_form_fields() method to return a list of form fields that you want to use in your form.
When you create a new Wagtail model, you can specify the custom model form class to use by setting the base_form_class attribute on your model. 
`from wagtail.admin.models import Page
from wagtail.admin.form import WagtailAdminModelForm

class MyPage(Page):
        base_form_class = MyCustomPageForm

class MyCustomPageForm(WagtailAdminModelForm):
        def get_form_fields(self):
        fields = super ().get_form_fields()
        fields["my_custom_field'] = MyCustomFormField()
        return fields`

**Comment by thibaudcolas** [comment]
@MrKeals I don’t understand how this fits with other documentation with this site? Did you report this on the wrong project?

**Comment by MrKeals** [comment]
Dear @thibaudcolas, Using the user guide alone via the link https://guide.wagtail.org/, I noticed the need to Use Custom Model Form Fields in Wagtail Admin Forms. I wasn't given access to the Wagtail Website but I used the the user guide as a point of reference for my deductions below. 
"The Ability to Use Custom Model Form Fields in Wagtail Admin Forms" section in the Wagtail documentation provides information on how to create customized user interfaces for editing Wagtail models. This can be useful for improving the accessibility of Wagtail admin forms, as it allows developers to create forms that are more user-friendly and intuitive for users with disabilities.

For example, a developer may use custom model form fields to add a dropdown list or a checkbox to a form, which can make it easier for users with mobility or visual impairments to select options or toggle settings. Additionally, custom form fields can be used to add new functionality to Wagtail admin forms, such as the ability to upload files or select multiple items from a list, which can be helpful for users with cognitive or learning disabilities.

Furthermore, using custom model form fields can save development time and effort when creating new Wagtail models or extending existing models, which can help to ensure that Wagtail's accessibility features are implemented consistently across all models and forms.

Overall, the "The Ability to Use Custom Model Form Fields in Wagtail Admin Forms" section in the Wagtail documentation can assist in the documentation process for accessibility on the Wagtail's User Guide by providing information on how to create customized, user-friendly, and accessible admin forms.
